embulk-standards/src/main/java/org/embulk/standards/RenameFilterPlugin.java

```diff
@@ -82,12 +90,361 @@ public PageOutput open(TaskSource taskSource, Schema inputSchema,
+        }
+        // TODO(dmikurube): Revisit this for better escaping.
+        if (passCharacters.contains("\\E")) {
+            throw new ConfigException("\"pass_characters\" in \"first_character_types\" must not contain \"\\E\"");
+        }
+
+        StringBuilder regexBuilder = new StringBuilder();
+        regexBuilder.append("^[^");
+        for (String target : passTypes) {
+            if (CHARACTER_TYPE_KEYWORDS.containsKey(target)) {
+                regexBuilder.append(CHARACTER_TYPE_KEYWORDS.get(target));
+            } else {
+                throw new ConfigException("\"" +target+ "\" is an unknown character type keyword");
+            }
+        }
+        if (!passCharacters.isEmpty()) {
+            regexBuilder.append("\\Q");
+            regexBuilder.append(passCharacters);
+            regexBuilder.append("\\E");
+        }
+        regexBuilder.append("].*");
+
+        Schema.Builder schemaBuidler = Schema.builder();
+        for (Column column : inputSchema.getColumns()) {
+            String name = column.getName();
+            if (name.matches(regexBuilder.toString())) {
+                if (replace.isPresent()) {
+                    name = replace.get() + name.substring(1);
+                }
+                else if (prefix.isPresent()) {
+                    name = prefix.get() + name;
+                }
+            }
+            schemaBuidler.add(name, column.getType());
+        }
+        return schemaBuidler.build();
+    }
+
+    private Schema applyLowerToUpperRule(Schema inputSchema) {
+        Schema.Builder builder = Schema.builder();
+        for (Column column : inputSchema.getColumns()) {
+            builder.add(column.getName().toUpperCase(Locale.ENGLISH), column.getType());
+        }
+        return builder.build();
+    }
+
+    private Schema applyTruncateRule(Schema inputSchema, TruncateRule rule) {
+        Schema.Builder builder = Schema.builder();
+        for (Column column : inputSchema.getColumns()) {
+            builder.add(column.getName().substring(0, rule.getMaxLength()), column.getType());
+        }
+        return builder.build();
+    }
+
+    private Schema applyUpperToLowerRule(Schema inputSchema) {
+        Schema.Builder builder = Schema.builder();
+        for (Column column : inputSchema.getColumns()) {
+            builder.add(column.getName().toLowerCase(Locale.ENGLISH), column.getType());
+        }
+        return builder.build();
+    }
+
+    private Schema applyRegexReplaceRule(Schema inputSchema, RegexReplaceRule rule) {
+        final String match = rule.getMatch();
+        final String replace = rule.getReplace();
+
+        Schema.Builder builder = Schema.builder();
+        for (Column column : inputSchema.getColumns()) {
+            // TODO(dmikurube): Check if we need a kind of sanitization?
+            builder.add(column.getName().replaceAll(match, replace), column.getType());
```

Comment:
Can we wrap `PatternSyntaxException` with `ConfigException`? `replaceAll` might throw the exception.